scrapers_next/ky/committees.py

```diff
@@ -0,0 +1,154 @@
+
+    def process_item(self, item):
+        href = item.find("a").get("href")
+        parent = get_parent(self.root, item)
+        return CommitteeDetail(
+            {"parent": parent, "chamber": "legislature"},
+            source=URL(href, timeout=30),
+        )
+
+
+class StatutoryCommitteeList(HtmlListPage):
+    source = "https://legislature.ky.gov/Committees/statutory-committee"
+    selector = CSS("[data-committeersn]")
+
+    def process_item(self, item):
+        href = item.find("a").get("href")
+        parent = get_parent(self.root, item)
+        return CommitteeDetail(
+            {"parent": parent, "chamber": "legislature"},
+            source=URL(href, timeout=30),
+        )
+
+
+class SpecialCommitteeList(HtmlListPage):
+    source = "https://legislature.ky.gov/Committees/special-committee"
+    selector = CSS("[data-committeersn]")
+
+    def process_item(self, item):
+        href = item.find("a").get("href")
+        parent = get_parent(self.root, item)
+        return CommitteeDetail(
+            {"parent": parent, "chamber": "legislature"},
+            source=URL(href, timeout=30),
+        )
+
+
+# Used by CommitteeList classes to get and item's parent committee name
+# Returns None if the item is not a subcommittee
+def get_parent(root, item):
+    parent_committee_id = item.get("data-parentcommitteersn")
+    parent = None
+    if parent_committee_id != "":
+        parent = XPath(
+            "//li[@data-committeersn='{id}']/a".format(id=parent_committee_id)
+        ).match(root)[0]
+        parent = strip_committee_name(parent.text_content())
+    return parent
+
+
+class CommitteeDetail(HtmlPage):
+    def process_page(self):
+        title = CSS(".committee-title > h2").match(self.root)[0].text_content()
+        parent_committee = self.input.get("parent")
+        com = get_committee_info(title, parent_committee)
+        com.add_source(self.source.url, note="committee details page")
+
+        # Make the scraper more fragile by doing an extra check to
+        # make sure expected chamber and detected chamber are the same
+        if com.chamber != self.input.get("chamber"):
+            raise Exception("Unexpected chamber")
+
+        # Add members, skipping if no members
+        try:
+            members = XPath("//ul[@class='member-list']/li/a").match(self.root)
+            for member in members:
+                add_member(com, member.text_content())
+        except Exception:
+            skip_when_no_members = False
+            if skip_when_no_members:
+                raise SkipItem("No committee members")
+        return com
+
+
+# Senate and House committee names end with " (S)" or " (H)"
+# Other committee names have no suffix
+def strip_committee_name(committee_name):
+    is_house_committee = committee_name.endswith(" (H)")
+    is_senate_committee = committee_name.endswith(" (S)")
+    if is_house_committee or is_senate_committee:
+        return committee_name[0:-4]
```

Comment:
`[0:-4]` may be hard to reverse engineer in the future. A small comment on the expected format of the object you're parsing would be helpful.

Reply:
A comment showing the expected format has been added with https://github.com/openstates/openstates-scrapers/pull/4413/commits/009ff96e45f2c233054883a0659ccd7d7510da6a .